### PR TITLE
Adding struct_to_map function

### PR DESCRIPTION
**Pull request by suvimh**
Here is the added function. It's the reverse of map_to_struct and has been working okay for me at least.

### CODE REVIEW

File: +toml/struct_to_map.m [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+%   values as `my_struct`, recursively converting struct values likewise. Any keys
+%   that are not already valid MATLAB identifiers will be preserved as is.
+
+function map = struct_to_map(struct)
+    if isstruct(struct)
+        map = containers.Map();
+        fields = fieldnames(struct);
+        for i = 1 : numel(fields)
+            key = fields{i};
+            value = NewPreferences.struct_to_map(struct.(key));
```
Comment (g-s-k) [inline +toml/struct_to_map.m:13]: Can you change the module name to `toml` here?

```suggestion
            value = toml.struct_to_map(struct.(key));
```

File: +toml/struct_to_map.m [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            map(key) = value;
+        end
+    elseif iscell(struct)
+        map = cell(size(struct));
+        for i = 1 : numel(struct)
+            map{i} = NewPreferences.struct_to_map(struct{i});
```
Comment (g-s-k) [inline +toml/struct_to_map.m:19]: Likewise here

```suggestion
            map{i} = toml.struct_to_map(struct{i});
```

**Review by g-s-k — CHANGES_REQUESTED** [review]
Thanks for submitting this! Just a small tweak and I think it is good to go.

File: +toml/struct_to_map.m (hunk 1 above)
Comment (suvimh) [reply +toml/struct_to_map.m]: Yea of course, the updated code has now been committed.

File: +toml/struct_to_map.m (hunk 2 above)
Comment (suvimh) [reply +toml/struct_to_map.m]: Fixed :)

**Pull request merged**